bayesian_cnn_prometheus/learning/bayesian_detector.py

```diff
@@ -0,0 +1,121 @@
+        self._valid_data = None
+        self._test_data = None
+
+        # For creating a model
+        self._kernel_size = config.get('kernel_size')
+        self._activation = config.get('activation')
+        self._padding = config.get('padding')
+        self._prior_std = config.get('prior_std')
+
+        self._initial_epoch = config.get('initial_epoch')
+        self._kl_start_epoch = config.get('kl_start_epoch')
+        self._kl_alpha_increase_per_epoch = config.get('kl_alpha_increase_per_epoch')
+        self._kl_alpha = self._adjust_kl_alpha(config.get('kl_alpha'))
+
+        self._weights_path = config.get('weights_path')
+        self._weights_dir = config.get('weights_dir')
+        self._checkpointer: ModelCheckpoint = None
+        self._scheduler: LearningRateScheduler = None
+        self._annealer: Callback = None
+        self._kl_start_epoch: int = config.get('kl_start_epoch')
+        self._kl_alpha_increase_per_epoch: float = config.get('kl_alpha_increase_per_epoch')
+
+        # For fitting
+        self._lr_decay_start_epoch = config.get('lr_decay_start_epoch')
+        self._epochs = config.get('epochs')
+        self._initial_epoch = config.get('initial_epoch')
+        self._valid_ds = config.get('valid_ds')
+
+    def _adjust_kl_alpha(self, kl_alpha: int):
+        if kl_alpha is None:
+            kl_alpha = 0.2  # TODO create a place to store default values// .2 is just a blind guess
+        if self._initial_epoch >= self._kl_start_epoch:
+            kl_alpha = min(1.,
+                           kl_alpha + (self._initial_epoch - self._kl_start_epoch) * self._kl_alpha_increase_per_epoch)
+
+        return K.variable(kl_alpha)
+
+    def fit(self, X, y):
+        print('Initializing the model...')
+        self._initialize_model(X)
+        self._initialize_callbacks()
+
+        print('Fitting the model...')
+        self._model.fit(x=X, epochs=self._epochs, initial_epoch=self._initial_epoch,
+                        callbacks=[self._checkpointer, self._scheduler, self._annealer],
+                        validation_data=y,
+                        steps_per_epoch=1, validation_steps=1)
+
+    def _initialize_model(self, X):
+        input_shape = BayesianDetector._get_input_shape(X)
+        train_len = self._calculate_train_len()
+
+        self._model = bayesian_vnet(input_shape, kernel_size=self._kernel_size, activation=self._activation,
+                                    padding=self._padding, prior_std=self._prior_std)
+        self._model.summary(line_length=127)
+        loss_function = variational_free_energy_loss(self._model, train_len / self._batch_size, self._kl_alpha)
+        self._model.compile(loss=loss_function, optimizer=Adam(), metrics=["accuracy"])
+
+    def _initialize_callbacks(self):
+        self._checkpoint_path = BayesianDetector._get_paths('bayesian', self._weights_dir)
+        self._checkpointer = ModelCheckpoint(self._checkpoint_path, verbose=1, save_weights_only=True,
```

Comment:
```suggestion
        self._checkpointer = ModelCheckpoint(str(self._checkpoint_path), verbose=1, save_weights_only=True,
```

Reply:
Później jest to formatowane jako string format, a Path nie może być "przekastowany" do str